packages/@react-spectrum/color/src/ColorEditor.tsx

```diff
@@ -0,0 +1,57 @@
+import {ColorArea} from './ColorArea';
+import {ColorField} from './ColorField';
+import {ColorSlider} from './ColorSlider';
+import {ColorSpace} from '@react-types/color';
+import {DOMRef} from '@react-types/shared';
+import {getColorChannels} from '@react-stately/color';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {Item, Picker} from '@react-spectrum/picker';
+import React, {CSSProperties, useState} from 'react';
+import {style} from '@react-spectrum/style-macro-s1' with {type: 'macro'};
+import {useDOMRef} from '@react-spectrum/utils';
+import {useLocalizedStringFormatter} from '@react-aria/i18n';
+
+export interface SpectrumColorEditorProps {
+  /** Whether to hide the alpha channel color slider and color field. */
+  hideAlphaChannel?: boolean
+}
+
+function ColorEditor(props: SpectrumColorEditorProps, ref: DOMRef<HTMLDivElement>) {
+  let [format, setFormat] = useState<ColorSpace | 'hex'>('hex');
+  let domRef = useDOMRef(ref);
+  let formatter = useLocalizedStringFormatter(intlMessages, '@react-spectrum/color');
+
+  return (
+    <div className={style({display: 'flex', flexDirection: 'column', gap: 4})()} ref={domRef}>
+      <div className={style({display: 'flex', gap: 4})()}>
+        <ColorArea colorSpace="hsb" xChannel="saturation" yChannel="brightness" />
+        <ColorSlider colorSpace="hsb" channel="hue" orientation="vertical" />
+        {!props.hideAlphaChannel &&
+          <ColorSlider channel="alpha" orientation="vertical" />
+        }
+      </div>
+      <div className={style({display: 'flex', gap: 4})()}>
+        <Picker aria-label="Color format" isQuiet width="size-700" menuWidth="size-1000" selectedKey={format} onSelectionChange={f => setFormat(f as typeof format)}>
+          <Item key="hex">{formatter.format('hex')}</Item>
+          <Item key="rgb">{formatter.format('rgb')}</Item>
+          <Item key="hsl">{formatter.format('hsl')}</Item>
+          <Item key="hsb">{formatter.format('hsb')}</Item>
+        </Picker>
+        {format === 'hex'
+          ? <ColorField isQuiet width="size-1000" aria-label={formatter.format('hex')} />
+          : getColorChannels(format).map(channel => (
+            <ColorField key={channel} colorSpace={format === 'hex' ? 'rgb' : format} channel={channel} isQuiet width="size-400" flex UNSAFE_style={{'--spectrum-textfield-min-width': 0} as CSSProperties} />
```

Comment:
You already filter out the special case where format === 'hex' at [Line 41](https://github.com/adobe/react-spectrum/pull/6272/files/6284aece9005065d4ac163c8394be36c26abf484#diff-f4419983fd216e15e3e9d79cf93d497159c1ffb2480774d499fcc43be6753c87R41):

```suggestion
            <ColorField key={channel} colorSpace={format} channel={channel} isQuiet width="size-400" flex UNSAFE_style={{'--spectrum-textfield-min-width': 0} as CSSProperties} />
```

Reply:
not good enough for typescript unfortunately. it still complains unless I repeat the check :/